java/src/com/google/idea/blaze/java/projectview/JavaLanguageLevelSection.java

```diff
@@ -25,51 +25,59 @@
 import com.google.idea.blaze.base.projectview.section.SectionParser;
 import com.intellij.pom.java.LanguageLevel;
 import javax.annotation.Nullable;
+import java.util.Locale;
 
 /** Section to force the java language level used */
 public class JavaLanguageLevelSection {
-  public static final SectionKey<Integer, ScalarSection<Integer>> KEY =
+
+  public static final SectionKey<LanguageLevel, ScalarSection<LanguageLevel>> KEY =
       SectionKey.of("java_language_level");
   public static final SectionParser PARSER = new JavaLanguageLevelParser();
 
   public static LanguageLevel getLanguageLevel(
       ProjectViewSet projectViewSet, LanguageLevel defaultValue) {
     return projectViewSet
         .getScalarValue(KEY)
-        .map(i -> getLanguageLevel(i, defaultValue))
         .orElse(defaultValue);
   }
 
   @Nullable
   @VisibleForTesting
-  static LanguageLevel getLanguageLevel(Integer level, @Nullable LanguageLevel defaultValue) {
-    LanguageLevel parsed = LanguageLevel.parse(level.toString());
-    return parsed != null ? parsed : defaultValue;
+  static LanguageLevel getLanguageLevel(String level, @Nullable LanguageLevel defaultValue) {
+    try {
+      return LanguageLevel.valueOf(level.toUpperCase(Locale.ROOT));
+    } catch (IllegalArgumentException e) {
+      // fall through
+    }
+
+    LanguageLevel languageLevel = LanguageLevel.parse(level);
```

Comment:
IIUC this part is for supporting the old format of java_language_level, if so we need to check that `level` is an Integer before calling `LanguageLevel.parse()`.

Reply:
I can add this check if you'd like but `LanguageLevel.parse` returns null for invalid language levels and this is the same assumption in the existing implementation.

Reply:
I also thought that using `LanguageLevel.parse` is enough. But I tried setting the language level to `INVALID_17_INVALID` for example and it was accepted as `JDK_17` that was surprising and that is why I want to force a separate validation here.

Reply:
i see, looking into the implementation of LanguageLevel.parse, it looks like it is pretty liberal in accepting version strings so adding this check will reject language levels that are currently accepted today in project view files. Seems like the right thing to do imo but just wanted to be clear about the implications.